napari/_qt/qt_viewer.py

```diff
@@ -433,6 +434,7 @@ def _create_canvas(self) -> None:
             vsync=True,
             parent=self,
             size=self.viewer._canvas_size[::-1],
+            autoswap=os.environ.get("NAPARI_AUTOSWAP"),  # see #5734
```

Comment:
```suggestion
            autoswap=os.environ.get("NAPARI_AUTOSWAP", "0") == "1",  # see #5734
```
Better is check for given water than presence.